core/libcamera_app.cpp

```diff
@@ -144,13 +171,71 @@ void LibcameraApp::CloseCamera()
 		LOG(2, "Camera closed");
 }
 
+Mode LibcameraApp::selectModeForFramerate(const libcamera::Size req, double fps)
+{
+	auto scoreFormat = [](double desired, double actual) -> double
+	{
+		double score = desired - actual;
+		// Smaller desired dimensions are preferred.
+		if (score < 0.0)
+			score = (-score) / 8;
+		// Penalise non-exact matches.
+		if (actual != desired)
+			score *= 2;
+
+		return score;
+	};
+
+	constexpr float penalty_AR = 1500.0;
+	constexpr float penalty_BD = 500.0;
+	constexpr float penalty_FPS = 2000.0;
```

Comment:
Hehe. I shall wait for the feature request to expose these on the command line! :)